### PR TITLE
Fix import error when `distributed` module not available

### PR DESCRIPTION
**Pull request by krshrimali**
## What does this PR do?

Fixes https://github.com/PyTorchLightning/pytorch-lightning/issues/12725.

Confirmed that this fixes it (built PyTorch from source with `USE_DISTRIBUTED=0`)

### Does your PR introduce any breaking changes? If yes, please list them.

No

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda

### CODE REVIEW

File: pytorch_lightning/strategies/ddp.py [hunk 1]
```diff
@@ -63,7 +63,8 @@
 if _FAIRSCALE_AVAILABLE:
     from fairscale.optim import OSS
 if _TORCH_GREATER_EQUAL_1_10:
```
Comment (akihironitta) [inline pytorch_lightning/strategies/ddp.py:65]: This is simpler maybe?
```suggestion
if _TORCH_GREATER_EQUAL_1_10 and torch.distributed.is_available():
```

**Review by akihironitta — APPROVED** [review]

File: pytorch_lightning/strategies/ddp.py (hunk 1 above)
Comment (krshrimali) [reply pytorch_lightning/strategies/ddp.py:65]: Ah, smart!!! @akihironitta - Thanks!

File: pytorch_lightning/strategies/ddp.py [hunk 2]
```diff
@@ -62,8 +62,9 @@
 
 if _FAIRSCALE_AVAILABLE:
     from fairscale.optim import OSS
-if _TORCH_GREATER_EQUAL_1_10:
-    from torch.distributed.algorithms.model_averaging.averagers import ModelAverager
+if _TORCH_GREATER_EQUAL_1_10 and torch.distributed.is_available():
+    if torch.distributed.is_available():
+        from torch.distributed.algorithms.model_averaging.averagers import ModelAverager
```
Comment (rohitgr7) [inline pytorch_lightning/strategies/ddp.py:67]:
```suggestion
if _TORCH_GREATER_EQUAL_1_10 and torch.distributed.is_available():
    from torch.distributed.algorithms.model_averaging.averagers import ModelAverager
```
??

**Review by rohitgr7 — APPROVED** [review]

**Review by Borda — APPROVED** [review]

File: pytorch_lightning/strategies/ddp.py (hunk 2 above)
Comment (krshrimali) [reply pytorch_lightning/strategies/ddp.py:67]: Good catch. Something went wrong with the previous suggestion. I didn't notice. Thanks!

File: pytorch_lightning/overrides/torch_distributed.py [hunk 3]
```diff
@@ -15,7 +14,10 @@
 logger = logging.getLogger(__name__)
 
 if torch.distributed.is_available():
+    from torch._C._distributed_c10d import ProcessGroup
     from torch.distributed import Backend, broadcast, get_backend, get_rank, GroupMember
+else:
+    ProcessGroup = None
```
Comment (awaelchli) [inline pytorch_lightning/overrides/torch_distributed.py:20]: ProcessGroup only gets imported for typing. Wouldn't the type check fail if the else branch is considered, as None is not a valid type annotation.

**Review by awaelchli — APPROVED** [review]

File: pytorch_lightning/overrides/torch_distributed.py (hunk 3 above)
Comment (krshrimali) [reply pytorch_lightning/overrides/torch_distributed.py:20]: I think `None` is a valid type annotation. Also, it doesn't error out on my system when `torch.distrubted.is_available()` is `False`. (source: https://stackoverflow.com/a/38516083)

```python
>>> def nothing(nun: None) -> None:
...     return nun
... 
>>> nothing(None)
>>> 
```

**Review by tchaton — APPROVED** [review]
LGTM !

File: pytorch_lightning/overrides/torch_distributed.py (hunk 3 above)
Comment (akihironitta) [reply pytorch_lightning/overrides/torch_distributed.py:20]: I think `None` may be too strict in this case. Should it be `object` instead? @awaelchli @krshrimali 
```suggestion
else:
    ProcessGroup = object
```

Comment (krshrimali) [reply pytorch_lightning/overrides/torch_distributed.py]: Thanks, @akihironitta for taking another look. I was just reading about it on the internet and found this interesting answer: https://stackoverflow.com/a/36286947. Apparently, this fixes for my environment (which doesn't have `distributed` module in PyTorch). What do you think? (just pushed a commit with my proposed solution)

**Review by carmocca — APPROVED** [review]

**Pull request merged**